packages/dara-core/js/auth/auth-wrapper.tsx

```diff
@@ -1,16 +1,54 @@
-/* eslint-disable react-hooks/exhaustive-deps */
-import { ReactNode, useContext, useEffect, useState } from 'react';
+import jwtDecode from 'jwt-decode';
+import { ReactNode, useContext, useEffect, useRef, useState } from 'react';
 import { Route, Switch } from 'react-router-dom';
 
+import { request } from '@/api/http';
 import DefaultFallback from '@/components/fallback/default';
 import ErrorPage from '@/pages/error-page';
 import Center from '@/shared/center/center';
 import { ImportersCtx } from '@/shared/context';
+import globalStore from '@/shared/global-state-store';
 import PrivateRoute from '@/shared/private-route/private-route';
 import { getToken, getTokenKey } from '@/shared/utils';
 
-import { AuthComponent, useAuthComponents } from './auth';
-import AuthContext from './auth-context';
+import { AuthComponent, useAuthConfig } from './auth';
+
+/**
+ * Refreshes the token if it's about to expire
+ *
+ * @param token token to refresh if expired
+ */
+async function checkTokenExpiry(token: string): Promise<string | null> {
+    // check if token is expired
+    if (token) {
+        try {
+            const decoded = jwtDecode<{
+                exp: number;
+            }>(token);
+
+            // If it's about to expire within 5 minutes
+            if (decoded.exp * 1000 - Date.now() < 5 * 60 * 1000) {
```

Comment:
I think 5 minutes is too long. Our target time for the token length is only 10minutes, so this will lead it to refreshing every 5 minutes which seems excessive. I'd be okay with either not having this at all as I think it overcomplicates it for not much benefit (as everything will now block and wait for the refresh anyway?) or reducing this timer down to 10-15 seconds?

Reply:
Sure, I agree the way things work now we can probably safely remove it and just let requests fail.